tket/src/Circuit/CircUtils.cpp

```diff
@@ -377,6 +380,110 @@ Circuit pauli_gadget(
   return circ;
 }
 
+void replace_CX_with_TK2(Circuit &c) {
+  static const Op_ptr cx = std::make_shared<Gate>(OpType::CX);
+  c.substitute_all(CircPool::CX_using_TK2(), cx);
+}
+
+Circuit with_TK2(Gate_ptr op) {
+  std::vector<Expr> params = op->get_params();
+  unsigned n = op->n_qubits();
+  if (n == 0) {
+    return Circuit();
+  } else if (n == 1) {
+    Circuit c(1);
+    c.add_op(op, std::vector<unsigned>{0});
+    return c;
+  } else if (n == 2 && op->free_symbols().empty()) {
+    Eigen::Matrix4cd U = op->get_unitary();
```

Comment:
Yeah, as part of the work I plan to do on two- and three-qubit squash I will add an option to this function to target TK2, and use that where appropriate. Not sure if we want to actually remove the CX decomposition method, since it is still useful in some contexts.